Allow customising shortcuts for activating next/previous service, see #53

insidewhy opened the pull request:
<!-- Thank you for your Pull Request. -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
<!-- Please start by naming your pull request properly for e.g. "Add Google Tasks to Todo providers". -->
<!-- Please keep in mind that any text inside "<!--" and "--\>" are comments from us and won't be visible in your bug report, so please don't put any text in them. -->

#### Pre-flight Checklist

Please ensure you've completed all of the following.

- [x] I have read the [Contributing Guidelines](https://github.com/ferdium/ferdium-app/blob/HEAD/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/ferdium/ferdium-app/blob/HEAD/CODE_OF_CONDUCT.md) that this project adheres to.

#### Description of Change
Allow customising certain shortcuts, for now just "activate next service" and "activate previous service".

Also changed default service switch shortcuts from `Cmd+Tab`/`Cmd+Shift+Tab` to `Ctrl+Tab`/`Ctrl+Shift+Tab` as the former key bindings are already bound by the OS and don't work, see https://github.com/ferdium/ferdium-app/pull/1920#issuecomment-2407203236

#### Motivation and Context
<!-- Why is this change required? What problem does it solve?  If it fixes an open issue, please link to the issue here. -->
Ctrl+tab and ctrl+shift+tab are not so ergonomic and this is an action that I use very frequently. This has been requested. Many people complained about these specific keyboard shortcuts in https://github.com/ferdium/ferdium-app/issues/53 and personally I prefer to use ctrl+, and ctrl+. for this.

#### Screenshots
<!-- Remove the section if this does not apply. -->

#### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [x] My pull request is properly named
- [x] The changes respect the code style of the project (`pnpm prepare-code`)
- [x] `pnpm test` passes
- [x] I tested/previewed my changes locally

#### Release Notes

Allow customising the keyboard shortcuts used to activate the next and previous services.

vraravam commented:
@insidewhy  - thanks for submitting the PR. Could you please also attach screenshots of the preference pane,  and maybe a small video on how this feature works?

insidewhy commented:
@vraravam I can capture a video of interacting with the preference dialogues but the changes to the keybindings can't be captured with a screen video unless I also video me using my computer keyboard

vraravam commented:
> @vraravam I can capture a video of interacting with the preference dialogues but the changes to the keybindings can't be captured with a screen video unless I also video me using my computer keyboard

np - I was able to test this functionality out in my laptop. For future reference, you can use an app like [keycastr](https://github.com/keycastr/keycastr) to capture key bindings and mouse clicks to demonstrate them

1 suggestion: for macos, the `Cmd+tab` is used to switch between running applications, and so I would suggest that you change the keybinding to be `Ctrl+tab` and `Ctrl+Shift+tab` for macos as well.

insidewhy commented:
@vraravam thanks for the suggestions. The default key bindings were taken from the current key bindings used by the app, so I didn't want to change them as part of this PR or it would impact the existing functionality of the app.

insidewhy commented:
@vraravam thanks for the suggestions. The default key bindings were taken from the current key bindings used by the app, so I didn't want to change them as part of this PR or it would impact the existing functionality of the app.

I suppose this collision between Mac system keybindings and the app has been an issue all this while. If you want to fix it along with this PR I'm happy to do so.

vraravam commented:
@insidewhy  - i noticed (after I posted the above comment), that I used to use `Option+Cmd+<left/right arrow>` for moving in forward/reverse direction of services. So, yes, i agree that this was a "clash" already pre-existing in ferdium.

That being said, yes, if you can please update the PR to fix this clash, let's have that bug fixed! Thanks!

vraravam reviewed (changes requested):
change requested as per prev comment. Also, please resolve conflicts by rebasing (not merging) from the `develop` branch

insidewhy commented:
> @insidewhy - i noticed (after I posted the above comment), that I used to use `Option+Cmd+<left/right arrow>` for moving in forward/reverse direction of services. So, yes, i agree that this was a "clash" already pre-existing in ferdium.
> 
> That being said, yes, if you can please update the PR to fix this clash, let's have that bug fixed! Thanks!

Just to make sure, you want the default binding for mac to be `⌘+tab` or `ctrl+tab`? Currently it's `Cmd+Tab`.

insidewhy commented:
@vraravam BTW I also noticed that changing the shortcuts does not require a restart, so I can remove that message.

insidewhy commented:
> change requested as per prev comment. Also, please resolve conflicts by rebasing (not merging) from the `develop` branch

Done

insidewhy committed:
Allow customising certain keyboard shortcuts

This allows customising next/previous service shortcuts, but opens the
door to using the same code and UI for customizing further shortcuts.

These particular shortcuts were requested to be customizable in the
github issue https://github.com/ferdium/ferdium-app/issues/53

vraravam reviewed (approved):
LGTM! Thanks for the contribution @insidewhy

vraravam commented:
@all-contributors please add @insidewhy for code

insidewhy commented:
@vraravam Thanks for your help and for merging this so quickly 🎄
